### PR TITLE
PR feature/security_sample (one working security test case)

### PR DESCRIPTION
**Pull request by marpuch**
Ok. This is it. The security module has been created. One test case is present as well. I kindly ask for review.

### CODE REVIEW

File: allure-framework-modules/allure-security-module/pom.xml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    <scm>
+        <connection>scm:git:git://github.com/devonfw/devonfw-testing.git</connection>
+        <developerConnection>scm:git:ssh://github.com:devonfw/devonfw-testing.git</developerConnection>
+        <url>http://github.com/devonfw/devonfw-testing/tree/master</url>
+    </scm>
+
+    <properties>
+        <project.build.sourceEncoding>UTF8</project.build.sourceEncoding>
+
+        <allure-core-module.version>4.12.0.4-SNAPSHOT</allure-core-module.version>
```
Comment (lstefaniszyn) [inline allure-framework-modules/allure-security-module/pom.xml]: You could switch to  released version =  4.12.0.4

File: allure-framework-modules/allure-security-module/src/main/java/com/capgemini/ntc/security/core/BasePage.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+	
+	static {
+		// Get analytics instance created in BaseTets
+		analytics = BaseTest.getAnalytics();
+		
+		// Get and then set properties information from selenium.settings file
+		propertiesFileSettings = createPropertiesSettings();
+		
+		// Read System or maven parameters
+		setRuntimeParametersSelenium();
```
Comment (lstefaniszyn) [inline allure-framework-modules/allure-security-module/src/main/java/com/capgemini/ntc/security/core/BasePage.java]: just rename to something like "setRuntimeParametersSecurity()"

File: allure-framework-modules/allure-security-module/src/main/java/com/capgemini/ntc/security/core/BasePage.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+	
+	@Override
+	public void onTestClassFinish() {
+		BFLogger.logDebug("BasePage.onTestClassFinish   " + this.getClass()
+		        .getSimpleName());
+	}
+	
+	@Override
+	public ModuleType getModuleType() {
+		return ModuleType.EXAMPLE;
```
Comment (lstefaniszyn) [inline allure-framework-modules/allure-security-module/src/main/java/com/capgemini/ntc/security/core/BasePage.java]: Please switch to ModuleType.SECURITY

File: allure-framework-modules/allure-security-module/src/main/java/com/capgemini/ntc/security/core/base/properties/PropertiesFileSettings.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
+package com.capgemini.ntc.security.core.base.properties;
+
+import com.google.inject.Inject;
+import com.google.inject.name.Named;
+
+public class PropertiesFileSettings {
```
Comment (lstefaniszyn) [inline allure-framework-modules/allure-security-module/src/main/java/com/capgemini/ntc/security/core/base/properties/PropertiesFileSettings.java]: Probably you do not use this class. If yes, then remove it please.

File: allure-framework-modules/allure-security-module/src/main/java/com/capgemini/ntc/security/core/base/runtime/RuntimeParameters.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+import com.capgemini.ntc.test.core.base.runtime.RuntimeParametersI;
+import com.capgemini.ntc.test.core.logger.BFLogger;
+
+/**
+ * This class stores various system properties These parameters are accessible while test case executes Example :
+ * PARAM_3("param_3", "1410") IS: mvn test -Dparam_3=1525 -Dtest=MyTestClass
+ * 
+ * @author LUSTEFAN
+ */
+public enum RuntimeParameters implements RuntimeParametersI {
```
Comment (lstefaniszyn) [inline allure-framework-modules/allure-security-module/src/main/java/com/capgemini/ntc/security/core/base/runtime/RuntimeParameters.java]: Probably you do not use this class. If yes, then remove it please.

File: allure-framework-modules/allure-security-module/src/resources/settings.properties [hunk 6]
```diff
@@ -0,0 +1,4 @@
+selenium.phantomjs=./lib/webdrivers/phantomjs/bin/phantomjs.exe
```
Comment (lstefaniszyn) [inline allure-framework-modules/allure-security-module/src/resources/settings.properties]: Probably you do not use this "selenium. ...."  parameters. If yes, then remove it please.

File: allure-framework-modules/allure-security-module/src/test/resources/settings.properties [hunk 7]
```diff
@@ -0,0 +1,2 @@
+name.property_1=This is value for prop 1
```
Comment (lstefaniszyn) [inline allure-framework-modules/allure-security-module/src/test/resources/settings.properties]: please remove this file

File: allure-app-under-test/src/test/java/com/capgemini/ntc/security/access/DirectoryBrowsingTest.java [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+ * needed and he certainly does not need to list files in the directories
+ * on your server (unless you deliberately desire this feature). Putting
+ * private content of your e.g. .git folder into your distribution package
+ * to be uploaded to the server and downloaded by the attacker because of
+ * directory browsing feature is a mortal sin on its own.
+ *
+ * @author Marek Puchalski, Capgemini
+ */
+@RunWith(Parameterized.class)
+public class DirectoryBrowsingTest extends SecurityPage {
```
Comment (lstefaniszyn) [inline allure-app-under-test/src/test/java/com/capgemini/ntc/security/access/DirectoryBrowsingTest.java]: Unfortunately  every "Test" file must extends from "BaseTest"  only then whole mechanism of Observers, Data Driven , Parametrized, Parallel, and so on,  are enabled. 

Could you please modify this.

File: allure-framework-modules/allure-security-module/src/main/java/com/capgemini/ntc/security/core/BasePage.java (hunk 3 above)
Comment (marpuch) [reply allure-framework-modules/allure-security-module/src/main/java/com/capgemini/ntc/security/core/BasePage.java]: This enumeration value is not present right now. Shall I extend it?

Comment (lstefaniszyn) [reply allure-framework-modules/allure-security-module/src/main/java/com/capgemini/ntc/security/core/BasePage.java]: Yes. Please extend it, and then update allure-core-module version to 4.12.0.5-SNAPSHOT

File: allure-app-under-test/src/main/java/com/capgemini/ntc/security/session/SessionManagerModule.java [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+/**
+ * Session manager configuration.
+ *
+ * @author Marek Puchalski, Capgemini
+ */
+public class SessionManagerModule extends AbstractModule {
+	
+	@Override
+	protected void configure() {
+		bind(ISessionManager.class).to(SessionManager.class); // Singleton
```
Comment (przwojtkow) [inline allure-app-under-test/src/main/java/com/capgemini/ntc/security/session/SessionManagerModule.java]: If no example, comments should be removed.

File: allure-app-under-test/src/resources/settings.properties [hunk 10]
```diff
@@ -1,5 +1,6 @@
 // Core Test module settings
-core.isAnalyticsEnabled = true
+core.isAnalyticsEnabled=true
+core.isEncryptionEnabled=true
 
 // Selenium module settings
 selenium.phantomjs=./lib/webdrivers/phantomjs/bin/phantomjs.exe
```
Comment (przwojtkow) [inline allure-app-under-test/src/resources/settings.properties]: Merge with develop. PhantomJS is no more supported.

**Comment by marpuch** [comment]
This should be it. Please look over again.

**Pull request merged**